shared/src/main/scala/coop/rchain/store/KeyValueTypedStoreSyntax.scala

```diff
@@ -46,3 +46,5 @@ final class KeyValueTypedStoreOps[F[_], K, V](
   def getOrElse(key: K, elseValue: V)(implicit f: Functor[F]): F[V] =
     get1(key).map(_.getOrElse(elseValue))
 }
+
+object KeyValueTypedStoreSyntaxObj extends KeyValueTypedStoreSyntax
```

Comment:
Why is this object definition required. Syntax trait should be extended in the project level package file.

https://github.com/rchain/rchain/blob/606c87d944ec7e4080387d0b8bd7b961e122992e/shared/src/main/scala/coop/rchain/shared/package.scala#L12-L17

So when used one import is needed for project extensions.
```scala
import coop.rchain.shared.syntax._
```

Reply:
Yes, I tried this way but it leads to errors such a
```
BlockProcessor.scala:51:56: value ||^ is not a member of type parameter F[Boolean]
[error]     val alreadyProcessed  = c.dagContains(b.blockHash) ||^ c.bufferContains(b.blockHash)
```
I guess the problem with this
```scala
trait AllSyntaxShared
    extends KeyValueStoreSyntax
    with KeyValueTypedStoreSyntax
    with KeyValueStoreManagerSyntax
    with MonixableSyntax
    with Fs2StreamSyntax
    with catscontrib.ToBooleanF  // <--
```

Reply:
You are right, `ToBooleanF` should not be added (yet) to shared syntax because it can cause duplicate import with (FYI @nzpr):
https://github.com/rchain/rchain/blob/94ac5f79beb51445586e86bd9c8c7854c1ef5118/shared/src/main/scala/coop/rchain/catscontrib/Catscontrib.scala#L1-L5

I have small commit to fix that.